modules/tensor_mechanics/src/userobjects/ElementPropertyReadFile.C

```diff
@@ -147,6 +176,20 @@ ElementPropertyReadFile::getElementData(const Elem * elem, unsigned int prop_num
   return _data[jelem * _nprop + prop_num];
 }
 
+Real
+ElementPropertyReadFile::getBlockData(const Elem * elem, unsigned int prop_num) const
+{
+  unsigned int elem_subdomain_id = elem->subdomain_id();
+  mooseAssert(elem_subdomain_id < _nblock,
```

Comment:
This should probably be a regular runtime check rather than an assertion that is only performed in debug mode. Performance is not critical here.

Reply:
Oh I see. Just to be sure, I should throw `mooseError`s here, right?

Reply:
or `paramError` if applicable!